frontend/imgconfig.go

```diff
@@ -0,0 +1,117 @@
+package frontend
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/dalec"
+	"github.com/containerd/containerd/platforms"
+	"github.com/moby/buildkit/client/llb/sourceresolver"
+	"github.com/moby/buildkit/frontend/dockerui"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type imageBuilderConfig struct {
+	platform *ocispecs.Platform
+}
+
+type ConfigOpt func(*imageBuilderConfig)
+
+func WithPlatform(p ocispecs.Platform) ConfigOpt {
+	return func(c *imageBuilderConfig) {
+		c.platform = &p
+	}
+}
+
+func BuildImageConfig(ctx context.Context, client gwclient.Client, spec *dalec.Spec, targetKey string, dflt string, opts ...ConfigOpt) (*dalec.DockerImageSpec, error) {
+	dc, err := dockerui.NewClient(client)
+	if err != nil {
+		return nil, err
+	}
+
+	builderCfg := imageBuilderConfig{}
+	for _, optFunc := range opts {
+		optFunc(&builderCfg)
+	}
+
+	baseImgRef := getBaseOutputImage(spec, targetKey, dflt)
+	platform := platforms.DefaultSpec()
+	if builderCfg.platform != nil {
+		platform = *builderCfg.platform
+	}
+
+	_, _, dt, err := client.ResolveImageConfig(ctx, baseImgRef, sourceresolver.Opt{
+		Platform: &platform,
+		ImageOpt: &sourceresolver.ResolveImageOpt{
+			ResolveMode: dc.ImageResolveMode.String(),
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error resolving image config: %w", err)
+	}
+
+	var img dalec.DockerImageSpec
+	if err := json.Unmarshal(dt, &img); err != nil {
+		return nil, fmt.Errorf("error unmarshalling image config: %w", err)
+	}
+
+	cfg := img.Config
+	if err := dalec.MergeImageConfig(&cfg, mergeSpecImage(spec, targetKey)); err != nil {
+		return nil, err
+	}
+
+	img.Config = cfg
+	return &img, nil
+}
+
+func getBaseOutputImage(spec *dalec.Spec, target, defaultBase string) string {
```

Comment:
Let's make this exported.
Then we can pass in the output of this as in input to `BuildImageConfig`, which removes a function paramerter for `BuildImageConfig`.

Reply:
done